src/Xamarin.Android.Build.Tasks/Tests/Xamarin.Android.Build.Tests/BuildTest.cs

```diff
@@ -3798,6 +3798,49 @@ public void LibraryProjectsShouldSkipGetPrimaryCpuAbi ()
 				Assert.IsTrue (b.Output.IsTargetSkipped (target), $"`{target}` should be skipped!");
 			}
 		}
+
+		[Test]
+		public void AllResourcesInClassLibrary ([Values (true, false)] bool useAapt2)
+		{
+			var path = Path.Combine ("temp", TestName);
+
+			// Create a "library" with all the application stuff in it
+			var lib = new XamarinAndroidApplicationProject {
+				ProjectName = "MyLibrary",
+				Sources = {
+					new BuildItem.Source ("Bar.cs") {
+						TextContent = () => "public class Bar { }"
+					},
+				}
+			};
+			lib.SetProperty ("AndroidApplication", "False");
+			lib.SetProperty ("AndroidUseAapt2", useAapt2.ToString ());
+
+			// Create an "app" that is basically empty and references the library
+			var app = new XamarinAndroidLibraryProject {
+				ProjectName = "MyApp",
+				Sources = {
+					new BuildItem.Source ("Foo.cs") {
+						TextContent = () => "public class Foo : Bar { }"
+					},
+				}
+			};
+			app.AndroidResources.Clear (); // No Resources
```

Comment:
should we be adding the App Resources (like Icon) to the library? This looks like there are no resources in the library either.

Reply:
I setup these tests so that:
* The "library" is an app project where I set `$(AndroidApplication)` to `False`
* The "app" is a library project where I set `$(AndroidApplication)` to `True`, and cleared `AndroidResource`. (It had `strings.xml`).